modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/identity/mgt/AccountLockEnabledTestCase.java

```diff
@@ -136,9 +144,92 @@ public void testSuccessfulEmailRetrieval() {
     public void atEnd() throws Exception {
         usmClient.deleteUser(testLockUser1);
         usmClient.deleteUser(testLockUser2);
+        usmClient.deleteUser(testLockUser3);
+        usmClient.deleteUser(testLockUser4);
+        usmClient.deleteRole(ACCOUNT_LOCK_BYPASS_ROLE);
         disableAccountLocking(ENABLE_ACCOUNT_LOCK);
     }
 
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.ALL})
+    @Test(groups = "wso2.is", description = "Check whether the user bypasses account locking successfully")
+    public void testLockByPassUnLockedAccount() {
+
+        try {
+            usmClient.addUser(testLockUser3, testLockUser3Password, new String[]{"admin"}, new ClaimValue[0], null, false);
+            usmClient.updateRoleListOfUser(testLockUser3, new String[]{"admin"}, new String[]{"admin", ACCOUNT_LOCK_BYPASS_ROLE});
+
+            int maximumAllowedFailedLogins = 5;
+            for (int i = 0; i < maximumAllowedFailedLogins; i++) {
+                try {
+                    authenticatorClient.login(testLockUser3, testLockUser3WrongPassword, "localhost");
+                } catch (Exception e) {
+                    log.error("Login attempt: " + i + " for user: " + testLockUser3 + " failed");
+                }
+            }
+
+            try {
+                authenticatorClient.login(testLockUser3, testLockUser3Password, "localhost");
+            } catch (Exception e) {
+                log.error("Login attempt for user: " + testLockUser3 + " failed");
+            }
+
+            ClaimValue[] claimValues = usmClient.getUserClaimValuesForClaims(testLockUser3, new String[]
+                    {accountLockClaimUri}, "default");
+
+            String userAccountLockClaimValue = null;
+
+            if (ArrayUtils.isNotEmpty(claimValues)) {
+                userAccountLockClaimValue = claimValues[0].getValue();
+            }
+
+            Assert.assertFalse
+                    ("Test Failure : User Account Didn't by pass locking", Boolean.valueOf(userAccountLockClaimValue));
+
+        } catch (Exception e) {
+            log.error("Error occurred when by passing the test user.", e);
+        }
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.ALL})
+    @Test(groups = "wso2.is", description = "Check whether the user bypasses account locking successfully " +
+            "for a locked account")
+    public void testLockByPassLockedAccount() {
+
+        try {
+            usmClient.addUser(testLockUser4, testLockUser4Password, new String[]{"admin"}, new ClaimValue[0], null, false);
+
+            int maximumAllowedFailedLogins = 5;
+            for (int i = 0; i < maximumAllowedFailedLogins; i++) {
+                try {
+                    authenticatorClient.login(testLockUser4, testLockUser4WrongPassword, "localhost");
+                } catch (Exception e) {
```

Comment:
No need to catch this exception

Reply:
Actually we need to the catch here, since I intentionally give wrong credentials, login method throws an exception. I need to continue after that point hence logged.